BuildTasks/Common/Common.ts

```diff
@@ -82,11 +82,19 @@ export function setTfxManifestArguments(tfx: ToolRunner): (() => void) {
         if (isPreview) { jsonOverrides.galleryFlags = ["Preview"]; };
     }
 
-    const extensionVersion = tl.getInput("extensionVersion", false);
+    let extensionVersion = tl.getInput("extensionVersion", false);
     if (extensionVersion) {
-        tl.debug(`Overriding extension version to: ${extensionVersion}`);
-        jsonOverrides = (jsonOverrides || {});
-        jsonOverrides.version = extensionVersion;
+       const extractedVersions = extensionVersion.match(/[0-9]+\.[0-9]+\.[0-9]+/);
+       if (extractedVersions && extractedVersions.length === 1) {
+            extensionVersion = extractedVersions[0];
+            tl.debug(`Overriding extension version to: ${extensionVersion}`);
+            jsonOverrides = (jsonOverrides || {});
+            jsonOverrides.version = extensionVersion;
+        }
+        else
```

Comment:
For now, I think it's better to keep the support for the 4th part, at least as long as it is supported by the Marketplace itself. People are using it and they know that it's supported by the gallery, so I think we should support it as well.

For the "Override task version", what happens if the "extensionVersion" variable has the value MyBuild_1.2.4 and the user decides to propagate the version to its build tasks? Today the regex used to extract the version parts  is looking for something that starts with the numbers. So I guess it won't work.